pkg/graveler/branch/protection_manager.go

```diff
@@ -36,39 +31,33 @@ func NewProtectionManager(settingManager *settings.Manager) *ProtectionManager {
 }
 
 func (m *ProtectionManager) GetRules(ctx context.Context, repository *graveler.RepositoryRecord) (*graveler.BranchProtectionRules, *string, error) {
-	rulesMsg, checksum, err := m.settingManager.GetLatest(ctx, repository, ProtectionSettingKey, &graveler.BranchProtectionRules{})
+	rulesMsg := &graveler.BranchProtectionRules{}
+	checksum, err := m.settingManager.GetLatest(ctx, repository, ProtectionSettingKey, rulesMsg)
 	if errors.Is(err, graveler.ErrNotFound) {
-		return &graveler.BranchProtectionRules{}, nil, nil
+		return &graveler.BranchProtectionRules{}, swag.String(""), nil
 	}
 	if err != nil {
 		return nil, nil, err
 	}
 	if proto.Size(rulesMsg) == 0 {
 		return &graveler.BranchProtectionRules{}, nil, nil
 	}
```

Comment:
In this case we have data, so we should return the checksum. Even if the data is an empty slice.

Reply:
Removed